platform/src/main/java/org/stellar/anchor/platform/service/Sep6MoreInfoUrlConstructor.java

```diff
@@ -0,0 +1,68 @@
+package org.stellar.anchor.platform.service;
+
+import java.net.URI;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.SneakyThrows;
+import org.apache.http.client.utils.URIBuilder;
+import org.stellar.anchor.SepTransaction;
+import org.stellar.anchor.auth.JwtService;
+import org.stellar.anchor.auth.Sep6MoreInfoUrlJwt;
+import org.stellar.anchor.platform.config.MoreInfoUrlConfig;
+import org.stellar.anchor.platform.config.PropertyClientsConfig;
+import org.stellar.anchor.sep24.MoreInfoUrlConstructor;
+import org.stellar.anchor.sep6.Sep6Transaction;
+import org.stellar.anchor.util.ConfigHelper;
+
+public class Sep6MoreInfoUrlConstructor extends MoreInfoUrlConstructor {
```

Comment:
Do we need to split `SimpleMoreInfoUrlConstructor` into SEP-6 and SEP-24 specific implementations? It doesn't look like there is anything specific to the SEPs here.

Reply:
No there's nothing specific between SEP6 and SEP24, but this constructor is specific to SEP6 and SEP24, distinct from the rest SEPs.

1. The functions `construct()` is tied to Sep6 or Sep24 txn (`getClientDomain`, `getSep10Account`, `getTransactionId`, `getClientDomain` are sep6 and sep24 txn only). One way to do it is that we extract all these outside and make `construct()` a more general function, or we need to do type check and casting everywhere.

2. We need to differentiate these 2 `MoreInfoUrlConstructor` when creating the beans or it will leads to unexpected behavior or errors when other beans want to reference them (e.g `Sep6Service` and `Sep24Service` need corresponding `MoreInfoUrlConstructor`). These might be achieved with `@Qualifier` annotation if we can merge these 2 `MoreInfoUrlConstructor` but I haven't try that.

Reply:
In this case, can we consider the factory pattern?
https://refactoring.guru/design-patterns/factory-method

Reply:
@JiahuiWho That makes sense, I'm ok with leaving it like this or using the factory pattern.

Reply:
I would prefer to do refactor in a separate PR and let this one serve it's purpose.